src/Http/Middleware/RateLimit.php

```diff
@@ -64,7 +64,8 @@ public function handle($request, Closure $next)
         $this->handler->rateLimitRequest($request, $route->getRateLimit(), $route->getRateLimitExpiration());
 
         if ($this->handler->exceededRateLimit()) {
-            throw new HttpException(403, 'You have exceeded your rate limit.', null, $this->getHeaders());
+            $retryAfter = $this->getHeaders()['X-RateLimit-Reset'] - time();
+            throw new TooManyRequestsHttpException($retryAfter, 'You have exceeded your rate limit.');
```

Comment:
Well you totally tweak exceptions the way you want...
If you are using Laravel, in `\App\Exceptions\Handler` you can do something like:

```php
/**
 * {@inheritdoc}
 */
protected function prepareException(Exception $e)
{
    $e = parent::prepareException($e);

    if ($e instanceof \Symfony\Component\HttpKernel\Exception\HttpException) {
        if ($e->getStatusCode() === 403 && $e->getMessage() === 'You have exceeded your rate limit.') {
            $headers = $e->getHeaders();
            $retryAfter = $headers['X-RateLimit-Reset'] - time();

            $e = new \Symfony\Component\HttpKernel\Exception\TooManyRequestsHttpException(($retryAfter, 'You have exceeded your rate limit.');
        }
    }

    return $e;
}
```

Then you can catch you `TooManyRequestsHttpException`...

Is that what you meant?